crates/utils/src/settings/structs.rs

```diff
@@ -57,26 +57,53 @@ pub struct PictrsConfig {
 #[derive(Debug, Deserialize, Serialize, Clone, SmartDefault, Document)]
 #[serde(default, deny_unknown_fields)]
 pub struct DatabaseConfig {
-  /// Username to connect to postgres
-  #[default("lemmy")]
-  pub(super) user: String,
-  /// Password to connect to postgres
-  #[default("password")]
-  pub password: String,
-  #[default("localhost")]
-  /// Host where postgres is running
-  pub host: String,
-  /// Port where postgres can be accessed
-  #[default(5432)]
-  pub(super) port: i32,
-  /// Name of the postgres database for lemmy
-  #[default("lemmy")]
-  pub(super) database: String,
+  #[serde(flatten, default)]
+  pub connection: DatabaseConnection,
+
   /// Maximum number of active sql connections
   #[default(5)]
   pub pool_size: usize,
 }
 
+#[derive(Debug, Deserialize, Serialize, Clone, SmartDefault, Document)]
+#[serde(untagged, deny_unknown_fields)]
+pub enum DatabaseConnection {
+  /// Configure the database by specifying a URI
+  ///
+  /// This is the preferred method to specify database connection details since
+  /// it is the most flexible. The given example uses peer authentication to
+  /// obviate the need for creating/configuring/managing passwords.
+  Uri {
+    /// Connection URI pointing to a postgres instance
+    #[doku(example = "postgresql:///lemmy?user=lemmy&host=/var/run/postgresql")]
+    uri: String,
+  },
+
+  /// Configure the database by specifying parts of a URI
+  ///
+  /// Note that [`DatabaseConnection::Uri`] should be preferred since it
+  /// provides greater control over how the connection is made. This merely
+  /// exists for backwards-compatibility.
```

Comment:
Im not convinced that this should be removed, it seems much easier for users who are unfamiliar with postgres. Why not keep both options?

Reply:
I'd argue it would be best to remove `Parts` because:

* It's less flexible than `Uri`
* Lemmy already has an environment variable for configuring the database via a connection URI, so there is already precedent for supporting connection URIs in Lemmy
* People should know how to write/use connection URIs anyway since they're going to need to know how to administrate PostgreSQL anyway, and most tools for this (e.g. `psql`) accept URIs
* It will reduce the amount of  "why can't I use the unix domain socket with the `Parts` configuration?" questions you'll get, since that's how I initially ran into this
* I'd guess that most smaller instances will have Lemmy and PostgreSQL running on the same machine, and `Uri` will allow them to use peer authentication which will save them the hassle of managing additional secrets

I'll change the doc comment to link to PostgreSQL's documentation for writing connection URIs though, which should help with the familiarity problem.